src/select/select.js

```diff
@@ -578,7 +556,9 @@ class Select extends React.Component<PropsT, SelectStateT> {
           <MultiValue
             value={value}
             key={`value-${i}-${value[this.props.valueKey]}`}
-            removeValue={() => {
+            removeValue={event => {
+              // stop propagation so that clicks on tag action button does not open dropdown
+              event.stopPropagation();
```

Comment:
It's better to avoid `stopPropagation` whenever possible since it can lead to [strange bugs](https://css-tricks.com/dangers-stopping-event-propagation/).

The better solution would be to implement this logic on the other side. So the dropdown would check if the event comes from `role=button` element  and if so, it would ignore it (our `Tag` component is now missing that role so it should be added first). 

It can be turned into more general function:

```js
export function checkIfInteractive(target: Element, rootElement: Element) {
  const DISABLED_ELEMENTS = [
    'input',
    'textarea',
    'select',
    'option',
    'button',
    'a'
  ];
  while (target !== rootElement) {
    if (DISABLED_ELEMENTS.includes(target.tagName.toLowerCase())) {
      return true;
    }
    const role = target.getAttribute('role');
    if (role === 'button' || role === 'link') {
      return true;
    }
    if (target.tagName) target = target.parentElement!;
  }
  return false;
}
```

Where the `rootElement` is the dropdown and `target` is the tag. Then if `checkIfInteractive` returns `true`, we can ignore it.

Reply:
updated - thanks